resources/fhir/binding_lookup.py

```diff
@@ -0,0 +1,90 @@
+import pathlib
+from typing import Iterator, List
+from zipfile import ZipFile
+
+import orjson
+import inflection
+import sqlite3
+
+
+def _class_property(element_id):
+    """Ensure type, property_name."""
+    parts = element_id.split('.')
+    if len(parts) == 2:
+        return f"{parts[0]}.{parts[1]}"  # {'type': ''.join(parts[0]), 'property_name': parts[1]}
+    else:
+        klass = [inflection.camelize(_) for _ in parts[:-1]]
+        return f"{''.join(klass)}.{parts[-1]}"  # {'type': ''.join(klass), 'property_name': parts[-1]}
+
+
+def _profile_in_bundle(bundle) -> Iterator[dict]:
+    """Generate profiles."""
+    for e in bundle.get('entry', []):
+        _ = e.get('resource', None)
+        if _:
+            yield _
+
+
+def _profile_in_files(file_names: List[str]) -> dict:
+    """Yield the elements with bindings."""
+    zip_file = next(iter([_ for _ in file_names if _.endswith('zip')]), None)
+    if zip_file:
+        with ZipFile(zip_file) as myzip:
+            for file_name in [_ for _ in file_names if not _.endswith('zip')]:
+                with myzip.open(file_name) as fp:
+                    for _ in _profile_in_bundle(orjson.loads(fp.read())):
+                        yield _
+    else:
+        for file_name in file_names:
+            with open(file_name) as fp:
+                for _ in _profile_in_bundle(orjson.loads(fp.read())):
+                    yield _
+
+
+def _element_in_profile(profile: dict) -> dict:
+    """Yield the elements with bindings."""
+    snapshot = profile.get('snapshot', None)
+    if snapshot:
+        for element in snapshot.get('element', []):
+            yield element
+
+
+def _elements_in_files(file_names):
+    """Yield the elements with bindings."""
+    for profile in _profile_in_files(file_names):
+        for element in _element_in_profile(profile):
+            yield element
+
+
+def _elements_with_bindings(file_names: List[str]) -> Iterator[dict]:
+    """Elements with bindings, keyed by Class.property"""
+    for element in _elements_in_files(file_names):
+        if 'binding' not in element:
+            continue
+        if element['binding'].get('valueSet', None):
+            yield {'key': _class_property(element['id']), 'element': element}
+
+
+def main():
+
+    generator = _elements_with_bindings(['5.0.0-definitions.json.zip', 'profiles-resources.json', 'profiles-others.json', 'profiles-types.json'])
+
+    path = pathlib.Path("~/.iceberg").expanduser()
+    path.mkdir(parents=True, exist_ok=True)
+    path = path / "element_bindings.sqlite"
+    connection = sqlite3.connect(path)
+
```

Comment:
Interesting approach of loading the FHIR schema bundle into a DB and then sharding the bundle out into individual resourceType schemas through SQL commands vs. just brute forcing the file with python.